test/ui-test/testSuites/global_shared/steps/startupSteps.py

```diff
@@ -48,10 +49,19 @@ def context_init(context):
 
     context.userData[_fixtures_root] = os.path.join(joined_path, "fixtures/") 
 
-def given_a_first_time_user_lands_on_and_generates_new_key(context):
+def given_a_first_time_user_lands_on(context):
     erase_directory(context.userData[_status_data_folder])
     start_application(context.userData[_aut_name])
     welcome_screen = StatusWelcomeScreen()
+    welcome_screen.agree_terms_and_conditions()
+
+def given_a_first_time_user_lands_on_with_data_folder(context, data_folder_path):
```

Comment:
Since we already have this step in `steps.py` --> `@Given("the user starts the application with a specific data folder \"|any|\"")`, you could update the code there and directly call this method, to try to just have only one implementation of the same thing. And also, be consistent with the method name and BDD definition. You can change the method name or the BDD. It is up to you, both options seem fine to me.

I.e: It is also done for example with `given_a_first_time_user_lands_on_and_generates_new_key` or with `given_a_first_time_user_lands_on_and_navigates_to_import_seed_phrase`.